src/mqtt.cpp

```diff
@@ -100,22 +100,32 @@ void mqttConnect() {
     mqttClient.connect();
 }
 
+//        
+
 void mqttLoop() {
-    if (millis() - mqPingRx > 20*MQ_TIMEOUT) {
-        printf("*** No MQTT response in %d seconds - resetting\n",  (millis()-mqPingRx)/1000);
+    uint32_t cachemqPingRx = mqPingRx;
```

Comment:
The volatile obviously should be used, but did not solve the problem by itself. The additional variable did the trick.

I also tried swapping the millis() and the MqPingRX in the test, but that did not solve it.